plugins/woocommerce-admin/client/products/fields/attribute-term-input-field/attribute-term-input-field.tsx

```diff
@@ -91,100 +100,155 @@ export const AttributeTermInputField: React.FC<
 	const selectedTermSlugs = ( value || [] ).map( ( term ) => term.slug );
 
 	return (
-		<SelectControl< ProductAttributeTerm >
-			items={ fetchedItems }
-			multiple
-			disabled={ disabled || ! attributeId }
-			label=""
-			getFilteredItems={ ( allItems ) => allItems }
-			onInputChange={ debouncedSearch }
-			placeholder={ placeholder || '' }
-			getItemLabel={ ( item ) => item?.name || '' }
-			getItemValue={ ( item ) => item?.slug || '' }
-			stateReducer={ ( state, actionAndChanges ) => {
-				const { changes, type } = actionAndChanges;
-				switch ( type ) {
-					case selectControlStateChangeTypes.ControlledPropUpdatedSelectedItem:
-						return {
-							...changes,
-							inputValue: state.inputValue,
-						};
-					case selectControlStateChangeTypes.ItemClick:
-						return {
-							...changes,
-							isOpen: true,
-							inputValue: state.inputValue,
-							highlightedIndex: state.highlightedIndex,
-						};
-					default:
-						return changes;
-				}
-			} }
-			selected={ value }
-			onSelect={ onSelect }
-			onRemove={ onRemove }
-			className="woocommerce-attribute-term-field"
-		>
-			{ ( {
-				items,
-				highlightedIndex,
-				getItemProps,
-				getMenuProps,
-				isOpen,
-			} ) => {
-				return (
-					<Menu isOpen={ isOpen } getMenuProps={ getMenuProps }>
-						{ [
-							isFetching ? (
-								<div
-									key="loading-spinner"
-									className="woocommerce-attribute-term-field__loading-spinner"
-								>
-									<Spinner />
-								</div>
-							) : null,
-							...items.map( ( item, menuIndex ) => {
-								const isSelected = selectedTermSlugs.includes(
-									item.slug
-								);
-
-								return (
-									<MenuItem
-										key={ `${ item.slug }` }
-										index={ menuIndex }
-										isActive={
-											highlightedIndex === menuIndex
-										}
-										item={ item }
-										getItemProps={ getItemProps }
+		<>
+			<SelectControl< ProductAttributeTerm >
+				items={ fetchedItems }
+				multiple
+				disabled={ disabled || ! attributeId }
+				label=""
+				getFilteredItems={ ( allItems, inputValue ) => {
+					if (
+						inputValue.length > 0 &&
+						! allItems.find(
```

Comment:
Just a note that this will fail if it returns an item that is falsy in any way, but it looks unlikely to cause any issues here.